pkg/scheduler/framework/runtime/framework.go

```diff
@@ -658,14 +687,28 @@ func (f *frameworkImpl) RunPreFilterExtensionAddPod(
 	podInfoToAdd *framework.PodInfo,
 	nodeInfo *framework.NodeInfo,
 ) (status *framework.Status) {
+	logger := klog.FromContext(ctx)
+	logger = klog.LoggerWithName(logger, "AddPod")
+	// We are adding LoggerWithValues for contextualized logging here
+	// but contextualized logging is feature gated and optional until
+	// it hits GA when it will be defaulted. For that reason
+	// logger.V(<LOGLEVEL>).Info or Error calls still need to include
+	// WithValues Key/Values while contextualized logging does not hit GA
+	// (duplicates get filtered out if contextualized logging is enabled)
+	// TODO(knelasevero): Remove duplicated keys from log entry calls
+	// When contextualized logging hits GA
+	// https://github.com/kubernetes/kubernetes/issues/111672
+	logger = klog.LoggerWithValues(logger, "pod", klog.KObj(podToSchedule), "node", klog.KObj(nodeInfo.Node()), "extensibleAPI", "PreFilter")
 	for _, pl := range f.preFilterPlugins {
 		if pl.PreFilterExtensions() == nil || state.SkipFilterPlugins.Has(pl.Name()) {
 			continue
 		}
+		logger := klog.LoggerWithName(logger, pl.Name())
+		ctx := klog.NewContext(ctx, logger)
 		status = f.runPreFilterExtensionAddPod(ctx, pl, state, podToSchedule, podInfoToAdd, nodeInfo)
 		if !status.IsSuccess() {
 			err := status.AsError()
-			klog.ErrorS(err, "Failed running AddPod on PreFilter plugin", "plugin", pl.Name(), "pod", klog.KObj(podToSchedule))
+			logger.Error(err, "Plugin failed", "pod", klog.KObj(podToSchedule), "node", klog.KObj(nodeInfo.Node()), "extensibleAPI", "PreFilter", "extension", "AddPod", "plugin", pl.Name())
```

Comment:
Can you provide an example what the log looks like?

Reply:
Currently, such an example has to produced locally, for example by starting a cluster with `local-up-cluster.sh`. 

@mengjiao-liu: do you know how to do that? Ping me on Slack if not.

Would it be useful to have an optional pull Prow job that could be triggered via `/test` to run some tests with contextual logging enabled? Probably yes.

Reply:
ok, I'll do it in a few days because of the holidays.

Reply:
In order to intuitively see the usage of `klog.LoggerWithName` and `klog.LoggerWithValues` in the kube-scheduler startup log, I directly modified 3 lines of code in the kube-scheduler `Run` function, the specific code is as follows:
```go
#cmd/kube-scheduler/app/server.go:153
	logger = klog.LoggerWithName(logger, "kube-scheduler-test-contextual-logging")
	logger = klog.LoggerWithValues(logger, "testContextualLogging", "kube-scheduler", "testDuplicateKey", "v1")
	// To help debugging, immediately log version
	logger.Info("Starting Kubernetes Scheduler", "version", version.Get(), "testDuplicateKey", "v2")

```
Run the following command locally:
```
hack/local-up-cluster.sh
```
output:
```
I0406 18:03:14.465697   51023 server.go:156] "kube-scheduler-test-contextual-logging: Starting Kubernetes Scheduler" testContextualLogging="kube-scheduler" version="v1.27.0-beta.0.58+2ec3a08e0bc4c4-dirty" testDuplicateKey="v2"
I0406 18:03:14.465730   51023 server.go:158] "kube-scheduler-test-contextual-logging: Golang settings" testContextualLogging="kube-scheduler" testDuplicateKey="v1" GOGC="" GOMAXPROCS="" GOTRACEBACK=""
```
Original output:
```
I0406 18:24:59.012179   60105 server.go:154] "Starting Kubernetes Scheduler" version="v1.27.0-beta.0.58+2ec3a08e0bc4c4-dirty"
I0406 18:24:59.012197   60105 server.go:156] "Golang settings" GOGC="" GOMAXPROCS="" GOTRACEBACK=""

```

log file: https://gist.github.com/mengjiao-liu/0b2eca9ab41e7727e7ae17d3e183d9e8

Would this log example be deemed satisfactory by reviewers? @kerthcet

Reply:
Ref https://github.com/kubernetes/kubernetes/pull/113994#issuecomment-1502997225
> Once https://github.com/kubernetes/test-infra/pull/29266 is merged, we can trigger jobs to produce logs with these changes.

Reply:
> I have a question about the log here is why `version="v1.27.0-beta.0.58+2ec3a08e0bc4c4-dirty"` is ahead of `testDuplicateKey="v2"`?

In short, it's the last place the duplicate key appears. In this example, the duplicate key `testDuplicateKey ` is after the key `version`.
```
	logger.Info("Starting Kubernetes Scheduler", "version", version.Get(), "testDuplicateKey", "v2")
```

Code located at：
```
// MergeKVs deduplicates elements provided in two key/value slices.
//
// Keys in each slice are expected to be unique, so duplicates can only occur
// when the first and second slice contain the same key. When that happens, the
// key/value pair from the second slice is used. The first slice must be well-formed
// (= even key/value pairs). The second one may have a missing value, in which
// case the special "missing value" is added to the result.
```
https://github.com/kubernetes/klog/blob/2fdc76f665b2e22daa8a8243c7c637315ecc0967/internal/serialize/keyvalues.go#L53-L60